### DISCUSSION TITLE
e2e-test shows insufficient funds

### DISCUSSION THREAD

**Discussion by RashRAJ** [hiero-ledger/hiero#4] [Q&A]
I am trying to work on this [issue](https://github.com/hashgraph/hedera-sdk-go/issues/1046) but when I run the command below to see the e2e test current behavior I get this error even though my balance is _Balance:94.18954228 ℏ_

` env OPERATOR_KEY=$OPERATOR_KEY OPERATOR_ID=$OPERATOR_ID go test -v -timeout 9999s -tags=e2e ./...` 

<img width="1008" alt="Screenshot 2024-10-04 at 02 07 02" src="https://github.com/user-attachments/assets/da377b68-c30d-41b0-b821-ee0179d44a5a">

**When I run** 
`env OPERATOR_KEY=$OPERATOR_KEY OPERATOR_ID=$OPERATOR_ID go test -v -timeout 9999s -tags=e2e ./utilities_for_test.go`

**I get this error, what could I be missing?**

<img width="506" alt="image" src="https://github.com/user-attachments/assets/33440b61-ab8e-4589-80a5-af9c2ba90c6a">

cc: @0xivanov @hendrikebbers

**Comment by 0xivanov** [comment]
Hey @RashRAJ 

You can try the following command to run the e2e tests:
```
env OPERATOR_KEY="<OPERATOR_KEY>" OPERATOR_ID="<OPERATOR_ID>" HEDERA_NETWORK="testnet" go test -tags="e2e" -v -timeout=9999s 
```

About the balance issues, when the testing env is created, a new operator account is created with it as well. This new operator has initial balance of 150 HBARs, meaning your initial operator account needs to have at least 150 HBARs. 
Note that when the testing env is closed, this newly created account is deleted and the remaining of it's balance is transferred back to your account.
If you want to change this for testing purposes, just edit line 129 in `utilities_for_test.go` (`SetInitialBalance(NewHbar(<balance>)).`)

**Comment by jwagantall** [comment]
@0xivanov @hendrikebbers @RashRAJ 
Is it ok to close this discussion?
